src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java

```diff
@@ -378,40 +386,63 @@ public void onOpened()
+                requestWrappers = getOpenRequestTreeOfCitizen();
+                return requestWrappers.size();
             }
-            else
+
+            @Override
+            public void updateElement(final int index, final Pane rowPane)
             {
-                final Image logo = findPaneOfTypeByID(DELIVERY_IMAGE, Image.class);
-                logo.setVisible(true);
-                logo.setImage(request.getDisplayIcon());
-            }
+                if (index < 0 || index >= requestWrappers.size())
+                {
+                    return;
+                }
+
+                final RequestWrapper wrapper = requestWrappers.get(index);
+                final Box wrapperBox = rowPane.findPaneOfTypeByID(WINDOW_ID_REQUEST_BOX, Box.class);
+                wrapperBox.setPosition(wrapperBox.getX() + 10*wrapper.getDepth(), wrapperBox.getY());
 
-            rowPane.findPaneOfTypeByID(REQUESTER, Label.class).setLabelText(request.getRequester().getDisplayName(request.getToken()).getFormattedText());
-            rowPane.findPaneOfTypeByID(REQUEST_SHORT_DETAIL, Label.class)
-              .setLabelText(request.getShortDisplayString().getFormattedText().replace("§f", ""));
+                final IRequest<?> request = wrapper.getRequest();
+                final ItemIcon exampleStackDisplay = rowPane.findPaneOfTypeByID(LIST_ELEMENT_ID_REQUEST_STACK, ItemIcon.class);
+                final List<ItemStack> displayStacks = request.getDisplayStacks();
 
-            request.getRequestOfType(IDeliverable.class).ifPresent((IDeliverable requestRequest) -> {
-                if (!isCreative && !InventoryUtils.hasItemInItemHandler(new InvWrapper(inventory), requestRequest::matches))
+                if (!displayStacks.isEmpty())
+                {
+                    exampleStackDisplay.setItem(displayStacks.get((lifeCount / LIFE_COUNT_DIVIDER) % displayStacks.size()));
+                }
+                else
+                {
+                    final Image logo = findPaneOfTypeByID(DELIVERY_IMAGE, Image.class);
+                    logo.setVisible(true);
+                    logo.setImage(request.getDisplayIcon());
+                }
+
+                ColonyView view = ColonyManager.getColonyView(citizen.getColonyId());
+                rowPane.findPaneOfTypeByID(REQUESTER, Label.class).setLabelText(request.getRequester().getDisplayName(view.getRequestManager(), request.getToken()).getFormattedText());
+                rowPane.findPaneOfTypeByID(REQUEST_SHORT_DETAIL, Label.class)
+                  .setLabelText(request.getShortDisplayString().getFormattedText().replace("§f", ""));
+
+                if (wrapper.getDepth() > 0)
                 {
                     rowPane.findPaneOfTypeByID(REQUEST_FULLFIL, ButtonImage.class).hide();
+                    rowPane.findPaneOfTypeByID(REQUEST_CANCEL, ButtonImage.class).hide();
+                }
+                else {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)